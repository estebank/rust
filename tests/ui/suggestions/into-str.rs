fn foo<'a, T>(_t: T) where T: Into<&'a str> {}

fn main() {
    foo(String::new());
    //~^ ERROR trait `From<String>` is not implemented for `&str`
}
