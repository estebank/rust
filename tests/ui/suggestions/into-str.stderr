error[E0277]: the trait `Into<&str>` is not implemented for `String`
  --> $DIR/into-str.rs:4:9
   |
LL |     foo(String::new());
   |     --- ^^^^^^^^^^^^^ the trait `Into<&str>` is not implemented for `String`, which is required by `String: Into<&str>`
   |     |
   |     required by a bound introduced by this call
   |
   = note: required for `String` to implement `Into<&str>`
note: required by a bound in `foo`
  --> $DIR/into-str.rs:1:31
   |
LL | fn foo<'a, T>(_t: T) where T: Into<&'a str> {}
   |                               ^^^^^^^^^^^^^ required by this bound in `foo`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
